app/src/main/java/com/simplecity/amp_library/model/Playlist.java

```diff
@@ -274,8 +274,7 @@ public boolean equals(Object o) {
 
         Playlist playlist = (Playlist) o;
 
-        if (id != playlist.id) return false;
-        return name != null ? name.equals(playlist.name) : playlist.name == null;
+        return id == playlist.id && (name != null ? name.equals(playlist.name) : playlist.name == null);
```

Comment:
I'd prefer not to make this change. 'Simplifying' the conditional statement generally makes it harder to read for equals() method.